GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,120 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep
+    {
+        public const string LooseObjects = "loose-objects.history";
+        private readonly string looseObjectsHistoryPath = null;
+        private readonly bool ignoreTimeRestriction = false;

```

Comment:
nit: don't set these `readonly` values here, so the compiler will check that they are initialized in the constructor.